src/linker.c

```diff
@@ -140,6 +135,8 @@ static jv jv_basename(jv name) {
 static jv find_lib(jq_state *jq, jv rel_path, jv search, const char *suffix, jv jq_origin, jv lib_origin) {
   if (jv_get_kind(search) != JV_KIND_ARRAY)
     return jv_invalid_with_msg(jv_string_fmt("Module search path must be an array"));
+  if (!jv_is_valid(rel_path))
+    return rel_path;
```

Comment:
Here, at 137, and 141, we probably need to `jv_free(search)` and maybe the other `jv` arguments...

Reply:
Granted, we already had a leak, so let's fix that separately.

Reply:
Good point. I wasn't thinking about it too hard since the existing early exits are pretty sloppy, but generally `find_lib` is going to want to free `rel_path` and `search` (only?).